src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -49,128 +51,276 @@ internal class CompletionItem : IComparable<CompletionItem>
-        /// the resultant text becomes "((int)foo).
-        /// </summary>
+        /// the resultant text becomes "((int)foo).</remarks>
         public TextSpan FilterSpan { get; }
 
         /// <summary>
-        /// A CompletionItem marked as a builder will be presented as an Intellisense Builder,
-        /// initially with its display text, which will be replaced as the user types.
+        /// The description of the item broken down into display parts.
+        /// If the description is empty, then the description can be accessed via <see cref="CompletionService.GetDescriptionAsync(Document, CompletionItem, CancellationToken)"/>.
+        /// </summary>
+        public ImmutableArray<SymbolDisplayPart> Description { get; }
+
+        /// <summary>
+        /// If true the <see cref="CompletionItem"/> represents the text that has been typed.
         /// </summary>
         public bool IsBuilder { get; }
 
         /// <summary>
-        /// When this property is true, the completion list will display a warning icon to the
-        /// right of the item's text, indicating that the corresponding symbol may not be
-        /// available in every project a linked file is linked into.
+        /// If true the completion list will display a warning icon.
         /// </summary>
-        /// <returns></returns>
         public bool ShowsWarningIcon { get; }
 
         /// <summary>
-        /// When this property is true, after performing the action associated with the item, 
-        /// formatting is performed on the change
+        /// If true the item is formatted after it has been committed.
         /// </summary>
-        public bool ShouldFormatOnCommit { get; internal set; }
-
-        public CompletionItemRules Rules { get; }
+        public bool ShouldFormatOnCommit { get; }
 
-        public CompletionItem(
-            CompletionListProvider completionProvider,
-            string displayText,
-            TextSpan filterSpan,
-            ImmutableArray<SymbolDisplayPart> description = default(ImmutableArray<SymbolDisplayPart>),
-            Glyph? glyph = null,
-            string sortText = null,
-            string filterText = null,
-            bool preselect = false,
-            bool isBuilder = false,
-            bool showsWarningIcon = false,
-            bool shouldFormatOnCommit = false,
-            CompletionItemRules rules = null)
-            : this(completionProvider, displayText, filterSpan,
-                   description.IsDefault ? (Func<CancellationToken, Task<ImmutableArray<SymbolDisplayPart>>>)null : c => Task.FromResult(description),
-                   glyph, /*hasAsyncDescription*/ false, sortText, filterText, preselect, isBuilder, showsWarningIcon, shouldFormatOnCommit, rules)
-        {
-        }
+        /// <summary>
+        /// True if the item represent an argument name in a named argument list.
+        /// </summary>
+        public bool IsArgumentName { get; }
```

Comment:
Should not be part of the services API.  controller should not know about this.  This is a language detail leaking into a language agnostic system.